lab-teddy/server.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const Client = require('./model/client');
+const net = require('net');
+const EE = require('events').EventEmitter;
+const ee = new EE();
+const server = net.createServer();
+const PORT = process.env.PORT || 8080;
+
+const pool = [];
+
+ee.on('default', (client, string) => {client.socket.write(`Not a valid command:${string.split(' ', 1)}`);
+});
+
+ee.on('@all', (client, string) => {
+  console.log('this is the string', string);
+  pool.forEach(c => c.socket.write(`${client.userName}: ${string}`));
+});
+
+ee.on('/nic', (client, string) => {
+  pool.forEach(c => {
+    if(string.trim() === c.nickName){
+      client.socket.write(`${string} already been used, please choose a new nickname.`);
+    }
+    c.socket.write(`${client.nickName} has changed their nickname to ${string}.`);
+  });
+  client.nickName = string.trim();
+});
+ee.on('@dm', (client, string) => {
+  pool.forEach(c => {
```

Comment:
You're going to have to use either a for loop with an if conditional, or the .find() method.    forEach WILL NOT break out on a condition(unless you throw an exception). It will continue iterating through the array, so this DM is not actually a dm. It is a broadcast.